docs/main/Solutioning/backup-and-restore/velero-br-fs.md

```diff
@@ -0,0 +1,314 @@
+
+### Install Velero with CSI Snapshot Support Enabled
+
+Install Velero (V1.13.2) with the `velero-plugin-for-csi` and with any S3 object storage. In this example, we are using the Google cloud bucket as our BackupStorageLocation. Refer to [Velero Documentation](https://velero.io/plugins/) for instructions on setting up the BackupStorageLocation.
+
+As an example we will be using Google Cloud Platform (GCP) and OpenEBS.
+
+1. Install Velero.
+
+```
+velero install --provider gcp --plugins velero/velero-plugin-for-gcp:v1.10.0,velero/velero-plugin-for-csi:v0.7.0  --use-node-agent --bucket velero-openebs --secret-file ./credentials-velero --use-volume-snapshots=true --features=EnableCSI
+```
+
+2. Once you have installed Velero, verify that Velero has installed correctly.
+
+**Command**
+
+```
+kubectl get pods -n velero
+```
+
+**Output**
+
+```
+NAME                      READY   STATUS    RESTARTS   AGE
+node-agent-9b9gz          1/1     Running   0          175m
+node-agent-dlrzt          1/1     Running   0          175m
+node-agent-q47cq          1/1     Running   0          175m
+velero-6b9b99494b-rkppw   1/1     Running   0          20h
+```
+
+3. Verify the backup and snapshot storage location.
+
+  - Use the following command to verify the backup:
+
+  **Command**
+
+  ```
+  kubectl get backupstoragelocation -n velero
+  ```
+
+  **Output**
+
+  ```
+  NAME      PHASE       LAST VALIDATED   AGE   DEFAULT
+  default   Available   34s              20h   true
+  ```
+
+  - Use the following command to verify the snapshot storage location:
+
+  **Command**
+
+  ```
+  velero snapshot-location get
+  ```
+
+  **Output**
+
+  ```
+  NAME      PROVIDER
+  default   gcp
+  ```
+
+### Deploy the Application with CSI Backed Volumes
+
+Before installing the application with CSI backed volumes, install the storage class and the volume snapshot class for the OpenEBS CSI driver by applying the below yaml to your cluster. Refer to the [Replicated PV Mayastor Configuration documentation](../../user-guides/replicated-storage-user-guide/replicated-pv-mayastor/rs-configuration.md#create-replicated-pv-mayastor-storageclasss) for more details.
+
+:::info
+Volume snapshots for multi-replicas volume is supported only from OpenEBS v4.1. The annotation `velero.io/csi-volumesnapshot-class: "true"` must be added to the volume snapshot class or the Velero backup will fail.
+:::
+
+**YAML**
+
+```
+---
+kind: VolumeSnapshotClass
+apiVersion: snapshot.storage.k8s.io/v1
+metadata:
+  name: csi-mayastor-snapshotclass
+  annotations:
+    velero.io/csi-volumesnapshot-class: "true"
+driver: io.openebs.csi-mayastor
+deletionPolicy: Delete
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: mayastor-3-thin
+parameters:
+  protocol: nvmf
+  repl: "3"
+  thin: "true"
+provisioner: io.openebs.csi-mayastor
+reclaimPolicy: Delete
+volumeBindingMode: Immediate
+```
+
+In this example, We have deployed a sample Nginx test application for backup and restore
```

Comment:
Should we mention something to this effect at the top?
Something like "In this guide, using Velero we will take a backup of a sample Nginx application with a Mayastor PV from a cluster,  transfer the backup to an object store and restore on a different cluster"